cgroups/fs/apply_raw.go

```diff
@@ -88,6 +91,93 @@ func Apply(c *cgroups.Cgroup, pid int) (map[string]string, error) {
 	return paths, nil
 }
 
+// Stop stops all processes in the current cgroup
+func Stop(c *cgroups.Cgroup) error {
+	d, err := getCgroupData(c, 0)
+	if err != nil {
+		return err
+	}
+
+	// first we send SIGTERM to all tasks inside the cgroup
+	if err := Kill(c, syscall.SIGTERM); err != nil {
+		return err
+	}
+
+	// if failed to stop with only SIGTERM, we send SIGKILL
+	if err := waitStop(d, 100*time.Millisecond, 5*time.Second); err != nil {
+		if err := Kill(c, syscall.SIGKILL); err != nil {
+			return err
+		}
+		return waitStop(d, 100*time.Millisecond, -1*time.Second)
```

Comment:
wait forever sounds dangerous. What if the process is stuck in the kernel? How will this impact users?